### ISSUE TITLE
VCC crashes with command line BIN parameter

### ISSUE THREAD

**Issue by tingtron** [VCCE/VCC#150]
When `Vcc.exe` is executed with a command line parameter for a BIN file, most of the time
- the Vcc application does not appear
- it turns out it crashes after 2-4 seconds.

However, once every 8-10 times, it runs the BIN program successfully.

I tried various ways of executing with the same behavior:
 - directly in Windows Command Line or from a different batch file or "shell"
 - using Run as Administrator
 - from within the VCC folder and from another folder
 - with or without any combination of carts and disks: with/without MPI cart, or with/without a disk, even with a disk having the same BIN file
 - with or without CPU over-clocking

When executed, the Task Manager shows two new copies of the VCC process with the same BIN argument, even though in all the methods of launching, only one launch was performed.
Their Status changes from Running to Suspended and then they vanish one after another within about 2-4 seconds.
![image](https://github.com/VCCE/VCC/assets/2383820/2bca3989-0726-4dd5-866a-5925a5c2568f)

Event Viewer reports mostly the following issues:
```
Faulting application name: Vcc.exe, version: 2.1.0.8, time stamp: 0x64c0d047
Faulting module name: Vcc.exe, version: 2.1.0.8, time stamp: 0x64c0d047
Exception code: 0xc0000005

Fault offset: 0x00019f40
Faulting process id: 0x0xC388
Faulting application start time: 0x0x1DA0962C72DC383
Faulting application path: c:\embed\Tandy\VCC\Vcc.exe
```

or very rarely
```
Fault offset: 0x000c4ea6
```

The BIN file is compiled with LWASM with a proper EXEC post-amble start address.
So it run without a problem from a disk
```
LOADM"SAMP1" : EXEC
```
SAMP1.ASM:
```
00050 	ORG $1200
00060 BEGIN	JMP	START
00070 	FDB	DONE-BEGIN
00100 START	LDA	#$F9
00110 	LDX	#$400
00120 SCREEN	STA	,X+
00130 	CMPX	#$600
00140 	BNE	SCREEN  
00150 WAIT	JSR	[$A000]
00160 	BEQ	WAIT
00170 	CLR	$71
00180 	JMP	[$FFFE]
00190 DONE	EQU	*
00200 	END BEGIN
```

![image](https://github.com/VCCE/VCC/assets/2383820/052e9701-c209-41e3-9299-88df35189530)
![image](https://github.com/VCCE/VCC/assets/2383820/d7ca01ca-a928-4592-bfdd-bfdbf252dd0e)

**Comment by tingtron** [comment]
Interesting workaround, which works mostly:
- use the `-d` extra parameter before the BIN positional parameter
```
...Tandy\src>"c:\embed\Tandy\VCC\VCC.exe" -d SAMP1.BIN
```

Then it comes up with a program running as expected.

**Comment by ejaquay** [comment]
Please try VCC-2.1.8.1

**Comment by tingtron** [comment]
It was a HackFest, not LazyAssFest.

**Comment by tingtron** [comment]
Yes, indeed when using the latest version VCC-2.1.8.1, that crashing issue with BIN command line parameter is resolved.
(I thought I downloaded recently. But I could've selected a previous release by mistake.)

But I am having another issue with the latest VCC-2.1.8.1:
- When selecting menu: File / Exit, the application hangs, instead of existing.
- When using the close [X] window frame icon, it exists normally.
And this File / Exit issue doesn't seem to reported. If it's not already addressed elsewhere, I can create an issue for it.

**Comment by klinems** [comment]
I also noticed the same issue when closing:When selecting menu: File / Exit, the application hangs, instead of existing.I just upgraded to the latest version using the install method.MichaelSent from my Galaxy
-------- Original message --------From: Ting Tron ***@***.***> Date: 10/28/23  22:35  (GMT-06:00) To: VCCE/VCC ***@***.***> Cc: Subscribed ***@***.***> Subject: Re: [VCCE/VCC] VCC crashes with command line BIN parameter (Issue #150) 
Yes, indeed when using the latest version VCC-2.1.8.1, that crashing issue with BIN command line parameter is resolved.
(I thought I downloaded recently. But I could've selected a previous release by mistake.)
But I am having another issue with the latest VCC-2.1.8.1:

When selecting menu: File / Exit, the application hangs, instead of existing.
When using the close [X] window frame icon, it exists normally.
And this File / Exit issue doesn't seem to reported. If it's not already addressed elsewhere, I can create an issue for it.

**Comment by ejaquay** [comment]
Yes please.  The idea of the new versioning system is to allow more
frequent bug fixes between enhancement updates.

**Comment by ejaquay** [comment]
This is fixed in 2.1.8.1   Created new issue for VCC hang when using File Exit.
